### ISSUE TITLE
Add CdmReference or name to CdmExecutionSettings

### ISSUE THREAD

**Issue by egillax** [OHDSI/Strategus#154]
I'm not able to find any way of inserting a reference or a name of the database to the executionSettings at runtime.  Previously we used the `connectionDetailsReference` for that purpose. This is something we need in modules related to PLP to display multi-database results in the shiny app properly. I know some institutions have the name and/or version of the database encoded in the schema but most sites don't so a way to add that specifically at runtime would be nice. 

My suggestion would be to add an optional argument `CdmReference` or `CdmName` to `CdmExecutionSettings`. Then we could feed it to PLP through it's modules. In PLP if this name is not specified it defaults to using the schema. I'm sure there are more use cases for this (any that need to view or store multi database results).

**Comment by anthonysena** [comment]
Hi @egillax - just thinking/working through this issue and wanted to post this up for your feedback.

One of the reasons why we don't provide a CDM name in the execution settings is that sometimes people will put a placeholder in this space vs. the proper name for their institution's CDM name. With this in mind, we wanted to source this information from the `CDM_SOURCE` table in the CDM (along with some other CDM attributes which we'll call "database metadata") to gather this instead of asking the user to provide it. That said, we don't expose this database metadata to modules - we only provide the database ID (that is constructed from the metadata). 

To that end, I'm planning to make an update in v1.0 to expose the `getCdmDatabaseMetaData` so users can see exactly what CDM database metadata is obtained and inspected outside of executing an analysis specification as shown below. This output would be part of the execution settings so you could potentially grab the `cdmSourceAbbreviation` as shown below. I think by including the CDM database metadata in the execution settings this would satisfy your request. Do you agree? 

``` r
library(Strategus)
#> Loading required package: CohortGenerator
#> Loading required package: DatabaseConnector
#> Loading required package: R6
#> 
#> Attaching package: 'Strategus'
#> The following object is masked from 'package:CohortGenerator':
#> 
#>     uploadResults
```

``` r
cdmExecutionSettings <- createCdmExecutionSettings(
  workDatabaseSchema = "main",
  cdmDatabaseSchema = "main",
  workFolder = ".",
  resultsFolder = "."
)
connectionDetails <- Eunomia::getEunomiaConnectionDetails()

databaseMetaData <- getCdmDatabaseMetaData(
  cdmExecutionSettings = cdmExecutionSettings,
  connectionDetails = connectionDetails
)
#> Connecting using SQLite driver
```

``` r

str(databaseMetaData)
#> 'data.frame':    1 obs. of  13 variables:
#>  $ cdmSourceName               : chr "Synthea synthetic health database"
#>  $ cdmSourceAbbreviation       : chr "Synthea"
#>  $ cdmHolder                   : chr "OHDSI Community"
#>  $ sourceDescription           : chr "SyntheaTM is a Synthetic Patient Population Simulator. The goal is to output synthetic, realistic (but not real"| __truncated__
#>  $ sourceDocumentationReference: chr "https://synthetichealth.github.io/synthea/"
#>  $ cdmEtlReference             : chr "https://github.com/OHDSI/ETL-Synthea"
#>  $ sourceReleaseDate           : Date, format: "2019-05-25"
#>  $ cdmReleaseDate              : Date, format: "2019-05-25"
#>  $ cdmVersion                  : chr "v5.3.1"
#>  $ vocabularyVersion           : chr "v5.0 18-JAN-19"
#>  $ cdmVersionConceptId         : num 1147638
#>  $ databaseId                  : int 85642205
#>  $ maxObsPeriodEndDate         : Date, format: "2019-07-03"
```

<sup>Created on 2024-08-19 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by egillax** [comment]
Hi @anthonysena,

Yes that will work perfectly thank you.

**Comment by anthonysena** [comment]
This is now implemented in the `v1.0-main` branch.
